Dev-2/Dev-2/StringAnalyzer.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Linq;
+
+namespace Dev_2
+{
+    public class StringAnalyzer
+    {
+
+        /// <summary>
+        /// Returns the maximum number of unequal consecutive characters in a string
+        /// </summary>
+        public int MaxOfNotIdenticalConsecutiveSymbols(string str)
+        {
+            int result = 1;
+            int buff = 1;
+
+            for (int i = 0; i < str.Length; i++) 
+            {
+                if( i < str.Length - 1 && str[i + 1] != str[i])
+                {
+                    buff++;
+                }
+                else
+                {
+                    if (buff > result)
+                    {
+                        result = buff;
+                    }
+                    buff = 1;
+                }
+            }
+            if (str == string.Empty)
+            {
+                return 0;
+            }
+            return result;                   
+        }
+
+        /// <summary>
+        /// Returns the maximum number of identical consecutive digits in a string
+        /// </summary>
```

Comment:
Much copy paste in three almost identical methods, try to add one common method for search and use it inside MaxOfIdenticalConsecutiveDigits and others